packages/@react-spectrum/s2/src/FullscreenDialog.tsx

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {ButtonGroupContext} from './ButtonGroup';
+import {composeRenderProps, OverlayTriggerStateContext, Provider, Dialog as RACDialog, DialogProps as RACDialogProps} from 'react-aria-components';
+import {ContentContext, HeaderContext, HeadingContext} from './Content';
+import {DOMRef} from '@react-types/shared';
+import {forwardRef} from 'react';
+import {Modal} from './Modal';
+import {style} from '../style' with {type: 'macro'};
+import {StyleProps} from './style-utils';
+import {useDOMRef} from '@react-spectrum/utils';
+
+// TODO: what style overrides should be allowed?
+export interface FullscreenDialogProps extends Omit<RACDialogProps, 'className' | 'style'>, StyleProps {
+  /**
+   * The variant of fullscreen dialog to display.
+   * @default "fullscreen"
+   */
+  variant?: 'fullscreen' | 'fullscreenTakeover',
+  /** Whether pressing the escape key to close the dialog should be disabled. */
+  isKeyboardDismissDisabled?: boolean
+}
+
+const heading = style({
+  gridArea: 'heading',
+  flexGrow: 1,
+  marginY: 0,
+  font: 'heading'
+});
+
+const header = style({
+  gridArea: 'header',
+  marginX: {
+    sm: 'auto'
+  },
+  font: 'body-lg'
+});
+
+const content =  style({
+  gridArea: 'content',
+  flexGrow: 1,
+  overflowY: {
+    default: 'auto',
+    // Make the whole dialog scroll rather than only the content when the height it small.
+    '@media (height < 400)': 'visible'
+  },
+  font: 'body'
+});
+
+const buttonGroup = style({
```

Comment:
Should this have some padding? 

![FullscreenDialog - Example ⋅ Storybook](https://github.com/user-attachments/assets/fd46abc5-193c-49ae-82c4-e34fa0921101)